### PR TITLE
Additional properties

### PR DESCRIPTION
**Pull request by boyko**
Hi, thanks for the great work! Maybe I am missing something, but when I tried to generate JSON schemas for models with nested map fields I noticed that currently these fields have a type = "array" schema. I changed the applyConverters function to output an array type or an object type with additionalProperties depending on the reflected type. I hope that this can be useful.

```typescript
class User {
  @IsString()
  name: string;
}

class Post {
  @Type(() => {
    return String
  })
  @MinLength(2, {each: true})
  userStatus: Map<string, string>
}

class PostWidthUsers {
  @ValidateNested({each: true})
  @Type(() => User)
  users: Map<string, User>
}

// Schemas
{
      "User": {
        "properties": {
          "name": {
            "type": "string"
          }
        },
        "type": "object",
        "required": [
          "name"
        ]
      },
      "Post": {
        "properties": {
          "userStatus": {
            "additionalProperties": {
              "minLength": 2,
              "type": "string"
            },
            "type": "object"
          }
        },
        "type": "object",
        "required": [
          "userStatus"
        ]
      },
      "PostWidthUsers": {
        "properties": {
          "users": {
            "additionalProperties": {
              "$ref": "#/definitions/User"
            },
            "type": "object"
          }
        },
        "type": "object",
        "required": [
          "users"
        ]
      }
    })
  }
```

### CODE REVIEW

**Comment by boyko** [comment]
Yes, it works without ValidateNested, I forgot to remove it before committing. Sorry. I have removed the comments and followed your suggestion in applyConverters. Thanks for your quick response!

**Comment by epiphone** [comment]
> Yes, it works without ValidateNested

What about without `@Type(() => string)`, can we afford to lose that one?

**Comment by boyko** [comment]
With the current implementation removing @Type(() => string) would result in an array type schema.

File: src/index.ts [hunk 1]
```diff
@@ -114,11 +114,28 @@ function applyConverters(
...
+    )
+    const isMap = typeMeta && new typeMeta.reflectedType() instanceof Map
+
     const converter =
       converters[meta.type] || converters[cv.ValidationTypes.CUSTOM_VALIDATION]
 
     const items = _.isFunction(converter) ? converter(meta, options) : converter
+
+    if (meta.each && isMap) {
+      // @ts-ignore
```
Comment (epiphone) [inline src/index.ts:131]: Can we get rid of this `ts-ignore`, perhaps with a type coercion if necessary?

**Comment by epiphone** [comment]
> With the current implementation removing @type(() => string) would result in an array type schema.

Okay. I was thinking ideally having `@Type` there for primitive types wouldn't be necessary since the string type can be inferenced from `@MinLength`, but we can get this merged as is and leave `@Type`-less inference for future work.

File: src/index.ts (hunk 1 above)
Comment (boyko) [reply src/index.ts]: Done.

**Comment by epiphone** [comment]
Published in `v2.0.2`. Thanks for the contribution & patience!

**Pull request merged**